### ISSUE TITLE
Library Redefinition

### ISSUE THREAD

**Issue by HoshinoTented** [aya-prover/aya-dev#759]
Assume that:
* A library `A`, which defines `prim I`
* A library `B`, which depends on library `A` and defines nothing.
* A library `C`, which depends on library `A` and library `B`, also defines nothing.

<details>
<summary>Aya said:</summary>

```plain
Compiling A
  [Info] Resolving source file dependency
    Done in 158ms
  [Tyck] Prim (Prim.aya)
I like these modules :)
    Library loaded in 412ms
Compiling A
  [Info] Resolving source file dependency
    Done in 1ms
  [Tyck] Prim (Prim.aya)
In file Prim.aya:1:5 ->

  1 | prim I
           ^^

Error: Redefinition of primitive `I'

Resolving interrupted due to:
1 error(s), 0 warning(s).
🔨
```

</details>

> See [here](https://github.com/HoshinoTented/aya-issues/tree/31f6d6603ea0f2f928f9928f203095634e19ce84) for the complete information.

**Comment by tsao-chi** [comment]
😨

**Comment by tsao-chi** [comment]
`A` should be only compiled once by the way

**Comment by HoshinoTented** [comment]
> We should compute SCC of dependencies before compiling the library.

The libraries won't depend on each other, so topo sorting may be enough.

**Comment by ice1000** [comment]
> > We should compute SCC of dependencies before compiling the library.
> 
> The libraries won't depend on each other

Are you sure? What if they declare so?

**Comment by imkiva** [comment]
> > We should compute SCC of dependencies before compiling the library.
> 
> The libraries won't depend on each other, so topo sorting may be enough.

But this is indeed caused by library dependency. And topo sorting is surely a method that computes SCC.

**Comment by imkiva** [comment]
Just relax, I am taking care of this bug.

**Comment by HoshinoTented** [comment]
> And topo sorting is surely a method that computes SCC.

That was amazing! I got it!

**Comment by ice1000** [comment]
Looks like it's closed by #978
